pkg/catalog/catalog.go

```diff
@@ -188,43 +119,18 @@ func ListDevfileComponents(registryName string) (DevfileComponentTypeList, error
 		return *catalogDevfileList, err
 	}
 
-	// 1. Load each devfile concurrently from the previously retrieved devfile index entries
-	// 2. Populate devfile components with devfile data
-	// 3. Add devfile component types to the catalog devfile list
-	retrieveDevfiles := util.NewConcurrentTasks(len(registryIndices))
-	devfileMutex := &sync.Mutex{}
+	// Add stacks to the catalog devfile list
 	for _, index := range registryIndices {
```

Comment:
If we remove the code block of downloading the devfile for getting information of devfile support criterias, we don't need this for loop either. `DevfileComponentType` population step can be done inside function `getDevfileIndexEntries` when we download the index file. With this for loop removal, we can increase the performance a lot as we don't need to iterate through the all index entries again.

Reply:
good catch, updated